### PR TITLE
Feature/search box form

### PR DESCRIPTION
**Pull request by bilalbutt044**

### CODE REVIEW

File: src/components/SearchBox.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+
+import { Wrapper } from "./GlobalStyles";
+
```
Comment (zsoltime) [inline src/components/SearchBox.js]: You don't use these, so should be removed.

File: src/components/SearchBox.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          setQuery("");
+        }}
+      >
+        <label>Start by searching for a phone number</label>
+        <input
+          type="text"
+          placeholder="Type a phone number"
+          onChange={e => setQuery(e.target.value)}
+          value={query}
+        />
```
Comment (zsoltime) [inline src/components/SearchBox.js]: It's not accessible. Either the `input` should be inside the `label` tag or you need to add an `id` on `input` and an `htmlFor` on the `label`.

File: src/components/SearchBox.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import styled from "styled-components";
+
+import { Wrapper } from "./GlobalStyles";
+
+import PropTypes from "prop-types";
+
+export const SearchBox = props => {
+  const [query, setQuery] = useState("");
+  return (
+    <main>
```
Comment (zsoltime) [inline src/components/SearchBox.js]: `main` is unnecessary here, please remove it.

File: src/pages/Home.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
-import React from 'react';
-import styled from 'styled-components';
+import React from "react";
+import styled from "styled-components";
 
-import { Wrapper } from '../components/GlobalStyles';
+import { Wrapper } from "../components/GlobalStyles";
+import { SearchBox } from "../components/SearchBox";
```
Comment (zsoltime) [inline src/pages/Home.js]: Not sure why you changed the quotes :)

File: src/pages/Home.js [hunk 5]
```diff
@@ -11,12 +12,16 @@ const Hero = styled.div`
   text-align: center;
 `;
 
+const Query = query => {
+  console.log(query);
+};
```
Comment (zsoltime) [inline src/pages/Home.js]: Function names are lowercase, components are capitalised.

File: src/pages/Home.js [hunk 6]
```diff
@@ -11,12 +12,16 @@ const Hero = styled.div`
...
 const Home = () => {
   return (
     <main>
       <Hero>
         <Wrapper>
           <h1>Home Page</h1>
+          <SearchBox onSubmit={Query} />
```
Comment (zsoltime) [inline src/pages/Home.js]: You don't really need a new function here, adding `onSubmit={console.log}` should be enough.

File: src/themes.js [hunk 7]
```diff
@@ -1,4 +1,15 @@
...
+  blue: "#3db8db",
+  darkBlue: "#186a81",
+  darkGreen: "#00a666",
+  green: "#00b777",
+  grey: "rgba(0, 0, 0, 0.2)",
+  lightBlue: "#e6ebef",
+  offWhite: "#f7f9f6",
+  lightPurple: "#f2eff7",
+  purple: "#6d60b0",
+  white: "#fff",
```
Comment (zsoltime) [inline src/themes.js]: Not sure, why you duplicated these :)

File: src/themes.js [hunk 8]
```diff
@@ -16,12 +27,12 @@ export default {
...
-  boxShadowLarge: '0 0.5rem 1rem rgba(0, 0, 0, 0.1)',
-  breakpointS: '600px',
-  breakpointM: '900px',
-  breakpointL: '1200px',
+  borderRadius: "0.25rem",
+  boxShadow: "0 0.375rem 0.75rem -0.5rem rgba(0, 0, 0, 0.1)",
+  boxShadowLarge: "0 0.5rem 1rem rgba(0, 0, 0, 0.1)",
+  breakpointS: "600px",
+  breakpointM: "900px",
+  breakpointL: "1200px",
```
Comment (zsoltime) [inline src/themes.js]: Again, the quote style's changed.

**Review by zsoltime — CHANGES_REQUESTED** [review]
Looks good, but there are a few styling issues and please read my other comments. It seems prettier doesn't work for you, or just ignores the `.prettierrc` file. Could you check it, please?

File: src/pages/Home.js (hunk 4 above)
Comment (bilalbutt044) [reply src/pages/Home.js]: `prettierrc` changed the quotes.

Comment (zsoltime) [reply src/pages/Home.js]: It's strange, it's single in here: https://github.com/codesect/cs01-team-grace-01/blob/cdf3f99cba6df5a0f458df68a56d255e2ef3cb77/.prettierrc#L11

Comment (bilalbutt044) [reply src/pages/Home.js]: in `master` branch it is applyign single quotes but not in `feature/seachboxform`  branch

File: src/pages/Home.js [hunk 9]
```diff
@@ -11,12 +12,16 @@ const Hero = styled.div`
   text-align: center;
 `;
 
+const query = query => {
+  console.log(query);
+};
```
Comment (zsoltime) [inline src/pages/Home.js]: As you've updated the onSubmit below, we no longer need this.

File: src/components/SearchBox.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+      onSubmit={e => {
+        e.preventDefault();
+        props.onSubmit(query);
+        setQuery('');
+      }}
+    >
+      <label>
+        Start by searching for a phone number
+        <input
+          type="text"
```
Comment (zsoltime) [inline src/components/SearchBox.js]: It should be either `type="search"` or `type="tel"`. Probably search is better. :)

File: src/themes.js [hunk 11]
```diff
@@ -6,8 +6,8 @@ const colors = {
   green: '#00b777',
   grey: 'rgba(0, 0, 0, 0.2)',
   lightBlue: '#e6ebef',
-  lightPurple: '#f2eff7',
   offWhite: '#f7f9f6',
+  lightPurple: '#f2eff7',
```
Comment (zsoltime) [inline src/themes.js]: You don't need to change the order of these. I think I changed it in one of my latest PRs. I don't really understand how merging the master haven't solved this.

**Review by zsoltime — CHANGES_REQUESTED** [review]
Getting closer :) Please check my comments.

File: src/themes.js (hunk 11 above)
Comment (bilalbutt044) [reply src/themes.js]: i dont know how this occure. anyway i manually shift `lightPurple` before `offWhite`

**Review by zsoltime — APPROVED** [review]
Great, will you try to add some tests, too? 😄

**Pull request merged**